packages/SidePanel/src/SidePanel.js

```diff
@@ -0,0 +1,190 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import Dialog from "./components/Dialog";
+import Footer from "./components/Footer";
+import Header from "./components/Header";
+import Overlay from "./components/Overlay";
+import Trigger from "./components/Trigger";
+import Group from "./components/Group";
+import FocusTrap from "./components/FocusTrap";
+import { extractChildren } from "./helpers";
+import useOffsetScroll from "./hooks/useOffsetScroll";
+import useBodyOverflow from "./hooks/useBodyOverflow";
+import useEscapeKey from "./hooks/useEscapeKey";
+
+const propTypes = {
+  /** The content for the SidePanel. */
+  children: PropTypes.node.isRequired,
+
+  /** The width of the open panel. */
+  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+  /** Callback triggered when the side panel needs to be close */
+  onClose: PropTypes.func,
+
+  /** Callback once the sidepanel has been opened event */
+  onAfterOpen: PropTypes.func,
+
+  /** Callback once the sidepanel has been closed event */
+  onAfterClose: PropTypes.func,
+
+  /** Control the visibility of the side panel. This prop makes the side panel */
+  isOpen: PropTypes.bool.isRequired,
+
+  /** Control the z position of the sidepanel */
+  zIndex: PropTypes.number,
+
+  /** Control y offset of the sidepanel */
+  offsetY: PropTypes.number,
+
+  /** Modify the look of the SidePanel */
+  kind: PropTypes.oneOf(["default", "child"]),
+
+  /** Disable scroll overflow on document.body when SidePanel is open. */
+  disableBodyOverflow: PropTypes.bool,
```

Comment:
Is this necessary? Can we just _always_ hide the document scrollbars when an overlay is used?

Reply:
🤔 good point. Not 100% about this though, can you confirm if you want this or just a suggestion?

Reply:
I'd like to try it without.  It's easier to add a new prop later, than deprecate one.